import "galaxy.proto";
import "master.proto";
import "agent.proto";
import "initd.proto";

package baidu.galaxy;

option cc_generic_services = true;

enum TraceLevel {
    TINFO = 0;
    TWARNING = 1;
    TERROR = 2;
}

message PodStat {
    // index
    optional string id = 1;  
    // index
    optional string jobid = 2;

    optional int32 cpu_used = 3;
    optional int64 mem_used = 4;
    optional int32 cpu_quota = 5;
    optional int64 mem_quota = 6;
    optional int32 dynamic_cpu_quota = 7;
    optional int64 dynamic_mem_quota = 8;
    optional int64 time = 9;
    optional string podid = 10;
    optional int64 read_bytes_ps = 11;
    optional int64 write_bytes_ps = 12;
    optional int64 syscr_ps = 13;
    optional int64 syscw_ps = 14;
}

message PodEvent {
    // index
    optional string id = 1; 
    // index
    optional string jobid = 2;
    // current stage
    optional PodStage stage = 3;
    // current state
    optional PodState state = 4;

    optional int64 time = 5;

    // log from master , scheduler or agent
    optional string from = 6;
    optional string reason = 7;
    // index
    optional string agent_addr = 8;
    optional string gc_dir = 9;
    optional string version = 10;
    // index
    optional TraceLevel level = 11;
    // index
    optional string podid = 12;
}


message JobStat {
    // index
    optional string id = 1;

    optional int32 replica = 2;
    optional int32 running = 3;
    optional int32 pending = 4;
    optional int32 death = 5;

    optional int64 cpu_quota = 6;
    optional int64 mem_quota = 7;

    optional int64 cpu_used = 8;
    optional int64 mem_used = 9;

    optional int64 dynamic_cpu_quota = 10;
    optional int64 dynamic_mem_quota = 11;

    optional int64 time = 12;
    optional string jobid = 13;
}

message JobEvent {
    // index
    optional string id = 1;
    // index
    optional string name = 2;
    // index
    optional string uname = 3;
    // index
    optional string action = 5;
    optional bytes desc = 6;
    optional string version = 7;
    optional JobState state = 8;
    optional JobUpdateState update_state = 9;
    optional int64 time = 10;
    // index
    optional TraceLevel level = 11;
    optional string jobid = 12;
}

message TaskEvent {
    // index
    optional string id = 1;
    // index
    optional string agent_addr = 2;
    // index
    optional string pod_id = 3;

    optional string initd_addr = 4;
    optional TaskState state = 5;
    optional TaskStage stage = 6;
    optional int64 ctime = 7;
    optional int64 ttime = 8;
    optional string task_chroot_path = 9;
    optional string error = 10;
    optional int32 exit_code = 11;
    optional string cmd = 12;
    // index
    optional TraceLevel level = 13;
    // index
    optional string job_id = 14;
    optional bool internal_error = 15;
    optional ProcessStatus deploy = 16; 
    optional ProcessStatus main = 17;
    optional string gc_dir = 18;
}

message AgentStat {
    optional string addr = 1;
    optional int64 job_count = 2;
    optional int64 cpu_millicores = 3;
    optional int64 cpu_used_millicores = 4;
    optional int64 mem_used = 5;
    optional int64 mem_quota = 6;

}


message ClusterStat {
    // index
    optional string data_center = 1;

    optional int64 total_node_count = 2;
    optional int64 alive_node_count = 3;
    optional int64 dead_node_count = 4;

    optional int64 total_job_count = 5;
    optional int64 scale_up_job_count = 6;
    optional int64 scale_down_job_count = 7;
    
    optional int64 total_cpu_millicores = 8;
    optional int64 total_cpu_used_millicores = 9;
    optional int64 total_cpu_assigned = 10;

    optional int64 total_memory = 11;
    optional int64 total_memory_used = 12;
    optional int64 time = 13;
    optional int64 total_pod_count = 14;
    optional int64 total_memory_assigned = 15;
    optional string id = 16;
}

message JobPodPair {
    optional string jobid = 1;
    optional string podid = 2;
}

message AgentEvent {
    // index pk
    optional string addr = 1;
    optional int64 time = 2;
    // index
    optional string action = 3;
    // index 
    optional string data_center = 4;
    optional string id = 5;
    repeated JobPodPair pods = 6;
} 

